saleor/discount/migrations/tasks/saleor3_17.py

```diff
@@ -187,3 +192,16 @@ def get_discount_voucher_id_to_code_map(model_discounts):
         voucher_id: code for voucher_id, code in vouchers.values_list("id", "code")
     }
     return voucher_id_to_code_map
+
+
+@app.task
+def set_promotion_rule_variants(start_id=None):
+    promotions = Promotion.objects.active()
+    kwargs = {"id__gt": start_id} if start_id else {}
+    rules = PromotionRule.objects.order_by("id").filter(
+        Exists(promotions.filter(id=OuterRef("promotion_id"))), **kwargs
+    )[:PROMOTION_RULE_BATCH_SIZE]
```

Comment:
```suggestion
    rules = (
        PromotionRule.objects.using(settings.DATABASE_CONNECTION_REPLICA_NAME)
        .order_by("id")
        .filter(
            Exists(
                promotions.using(settings.DATABASE_CONNECTION_REPLICA_NAME).filter(
                    id=OuterRef("promotion_id")
                )
            ),
            **kwargs
        )[:PROMOTION_RULE_BATCH_SIZE]
    )
```
This could use a replica, right? We only do fetch the IDs (to refetch batch for update from writer)

Reply:
Done